### DISCUSSION TITLE
Is RTK Query related to browser cache?

### DISCUSSION THREAD

**Discussion by dev-splin** [reduxjs/redux-toolkit#3467] [Q&A] [answered]
React 18 + Next 13 + RTK Query is being used.

In our project, we initialize the global data as below.
1. After calling api from getInitialProps in _app, put the result value into redux.
2. Proceed with hydrate in apiSlice and createSlice.
3. Use the redux data that contains the initialization data.

However, after a long time running the project, the data was not included in the redux, so I refreshed Browser Cache and it was executed normally.

When I searched, I found the above phrase in the two links below.

---
- [https://redux-toolkit.js.org/rtk-query/usage/persistence-and-rehydration](https://redux-toolkit.js.org/rtk-query/usage/persistence-and-rehydration)
> INFO
Generally, persisting API slices is not recommended and instead, mechanisms like Cache-Control Headers should be used in browsers to define cache behaviour. Persisting and rehydrating an api slice might always leave the user with very stale data if the user has not visited the page for some time. Nonetheless, in environments like Native Apps, where there is no browser cache to take care of this, persistance might still be a viable option.

- [https://redux-toolkit.js.org/rtk-query/usage/server-side-rendering](https://redux-toolkit.js.org/rtk-query/usage/server-side-rendering)
> While memory leaks are not anticipated, once a render is sent to the client and the store is being removed from memory, you may wish to also call store.dispatch(api.util.resetApiState()) to ensure that no rogue timers are left running.
---

But I didn't know the exact reason, so I asked this question.
Is RTK Query really related to browser cache?
Please reply. Thank you :)

**Comment by phryneas** [comment, marked as answer]
No, it's not related to the browser cache. It's an in-memory cache that holds onto your data so you don't need to refetch data everytime your component rerenders.
